packages/@glimmer/runtime/lib/compiled/opcodes/component.ts

```diff
@@ -116,21 +116,20 @@ APPEND_OPCODES.add(Op.IsComponent, vm => {
 APPEND_OPCODES.add(Op.CurryComponent, (vm, { op1: _meta }) => {
   let stack = vm.stack;
 
+  let definition = check(stack.pop(), CheckReference);
   let args = check(stack.pop(), CheckArguments);
   let captured: Option<ICapturedArguments> = null;
 
   if (args.length) {
     captured = args.capture();
-    args.clear();
   }
 
   let meta = vm.constants.getSerializable<TemplateMeta>(_meta);
   let resolver = vm.constants.resolver;
-  let definition = check(stack.pop(), CheckReference);
 
-  stack.push(new CurryComponentReference(definition, resolver, meta, captured));
+  vm.loadValue(Register.v0, new CurryComponentReference(definition, resolver, meta, captured));
 
-  expectStackChange(vm.stack, -args.length - 1, 'CurryComponent');
+  // expectStackChange(vm.stack, -args.length - 1, 'CurryComponent');
```

Comment:
This likely needs to be brought back and incorporate the new `pop` from the stack.